dbfit-java/core/src/main/java/dbfit/api/AbstractDbEnvironment.java

```diff
@@ -117,23 +116,41 @@ public final PreparedStatement createStatementWithBoundFixtureSymbols(
                 cs.setObject(i + 1, value);
             }
         }
-        return cs;
+
+        return createPreparedStatement(cs);
     }
 
     @Override
-    public DdlStatementExecution createDdlStatementExecution(String ddl)
+    public DdlStatement createDdlStatement(String ddl)
             throws SQLException {
-        return new DdlStatementExecution(getConnection().createStatement(), ddl);
+        return new DdlStatement(getConnection().createStatement(), ddl);
+    }
+
+    /**
+     * Create a {@link DbStatement} for the given prepared statement.
+     * This is the method to override if custom implementation needs to be returned
+     */
+    protected DbStatement createPreparedStatement(PreparedStatement statement) {
```

Comment:
This method name still seems strange as it takes a `PreparedStatement` as a parameter.